### PR TITLE
Add "show fex" parser for NXOS

### PR DESCRIPTION
**Pull request by virrsa**
## Description
Adds a parser for "show fex" for NXOS devices to display basic FEX information.

## Motivation and Context
Parser was needed to gather FEX information from our NXOS devices, and there was no parser to gather the information.

## Impact (If any)
N/A

## Screenshots:
```
python3 folder_parsing_job.py -o nxos -c ShowFex
2022-03-08T14:13:29: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:29: %AETEST-INFO: |                   Starting testcase SuperFileBasedTesting                    |
2022-03-08T14:13:29: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:29: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:29: %AETEST-INFO: |                            Starting section setup                            |
2022-03-08T14:13:29: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: The result of section setup is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: The result of testcase SuperFileBasedTesting is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: |                            Starting testcase nxos                            |
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: |                            Starting section setup                            |
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: The result of section setup is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: |                           Starting section ShowFex                           |
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: :                       Starting STEP 1: nxos -> ShowFex                       :
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: :              Starting STEP 1.1: Test Golden -> nxos -> ShowFex               :
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: :        Starting STEP 1.1.1: Gold -> nxos -> ShowFex -> golden_output         :
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: The result of STEP 1.1.1: Gold -> nxos -> ShowFex -> golden_output is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: The result of STEP 1.1: Test Golden -> nxos -> ShowFex is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: :               Starting STEP 1.2: Test Empty -> nxos -> ShowFex               :
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: :        Starting STEP 1.2.1: Empty -> nxos -> ShowFex -> empty_output         :
2022-03-08T14:13:30: %AETEST-INFO: +..............................................................................+
2022-03-08T14:13:30: %AETEST-INFO: The result of STEP 1.2.1: Empty -> nxos -> ShowFex -> empty_output is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: The result of STEP 1.2: Test Empty -> nxos -> ShowFex is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: The result of STEP 1: nxos -> ShowFex is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: +..........................................................+
2022-03-08T14:13:30: %AETEST-INFO: :                       STEPS Report                       :
2022-03-08T14:13:30: %AETEST-INFO: +..........................................................+
2022-03-08T14:13:30: %AETEST-INFO: STEP 1 - nxos -> ShowFex                          Passed    
2022-03-08T14:13:30: %AETEST-INFO: STEP 1.1 - Test Golden -> nxos -> ShowFex         Passed    
2022-03-08T14:13:30: %AETEST-INFO: STEP 1.1.1 - Gold -> nxos -> ShowFex -> golden_outputPassed    
2022-03-08T14:13:30: %AETEST-INFO: STEP 1.2 - Test Empty -> nxos -> ShowFex          Passed    
2022-03-08T14:13:30: %AETEST-INFO: STEP 1.2.1 - Empty -> nxos -> ShowFex -> empty_outputPassed    
2022-03-08T14:13:30: %AETEST-INFO: ............................................................
2022-03-08T14:13:30: %AETEST-INFO: The result of section ShowFex is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: |                           Starting section cleanup                           |
2022-03-08T14:13:30: %AETEST-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %AETEST-INFO: The result of section cleanup is => PASSED
2022-03-08T14:13:30: %AETEST-INFO: The result of testcase nxos is => PASSED
2022-03-08T14:13:30: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %GENIE-INFO: |                               Unittest results                               |
2022-03-08T14:13:30: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %GENIE-INFO:  SECTIONS/TESTCASES                                                      RESULT   
2022-03-08T14:13:30: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-08T14:13:30: %GENIE-INFO: .
2022-03-08T14:13:30: %GENIE-INFO: |-- SuperFileBasedTesting                                                 PASSED
2022-03-08T14:13:30: %GENIE-INFO: |   `-- setup                                                             PASSED
2022-03-08T14:13:30: %GENIE-INFO: `-- nxos                                                                  PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |-- setup                                                             PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |-- ShowFex                                                           PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |   |-- Step 1: nxos -> ShowFex                                       PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |   |-- Step 1.1: Test Golden -> nxos -> ShowFex                      PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |   |-- Step 1.1.1: Gold -> nxos -> ShowFex -> golden_output          PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |   |-- Step 1.2: Test Empty -> nxos -> ShowFex                       PASSED
2022-03-08T14:13:30: %GENIE-INFO:     |   `-- Step 1.2.1: Empty -> nxos -> ShowFex -> empty_output          PASSED
2022-03-08T14:13:30: %GENIE-INFO:     `-- cleanup                                                           PASSED
2022-03-08T14:13:30: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %GENIE-INFO: |                                   Summary                                    |
2022-03-08T14:13:30: %GENIE-INFO: +------------------------------------------------------------------------------+
2022-03-08T14:13:30: %GENIE-INFO:  Number of ABORTED                                                            0 
2022-03-08T14:13:30: %GENIE-INFO:  Number of BLOCKED                                                            0 
2022-03-08T14:13:30: %GENIE-INFO:  Number of ERRORED                                                            0 
2022-03-08T14:13:30: %GENIE-INFO:  Number of FAILED                                                             0 
2022-03-08T14:13:30: %GENIE-INFO:  Number of PASSED                                                             2 
2022-03-08T14:13:30: %GENIE-INFO:  Number of PASSX                                                              0 
2022-03-08T14:13:30: %GENIE-INFO:  Number of SKIPPED                                                            0 
2022-03-08T14:13:30: %GENIE-INFO:  Total Number                                                                 2 
2022-03-08T14:13:30: %GENIE-INFO:  Success Rate                                                            100.0% 
2022-03-08T14:13:30: %GENIE-INFO: --------------------------------------------------------------------------------
2022-03-08T14:13:30: %GENIE-INFO:  Total Passing Unittests                                                      2 
2022-03-08T14:13:30: %GENIE-INFO:  Total Failed Unittests                                                       0 
2022-03-08T14:13:30: %GENIE-INFO:  Total Errored Unittests                                                      0 
2022-03-08T14:13:30: %GENIE-INFO:  Total Unittests                                                              2 
2022-03-08T14:13:30: %GENIE-INFO: --------------------------------------------------------------------------------
```

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/nxos/show_fex.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    ''' Parser for "show fex"'''
+
+    cli_command = 'show fex'
+
+    def cli(self, output=None):
+        if output is None:
+            # Get output from device
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
```
Comment (danielgraziano) [inline src/genie/libs/parser/nxos/show_fex.py:42]:
```suggestion
        if output is None:
            output = self.device.execute(self.cli_command)
```
then on line 48 you can do:
``` python
for line in output.splitlines():

File: src/genie/libs/parser/nxos/show_fex.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        else:
+            out = output 
+        # Initial return dictionary
+        parsed_dict = {}
+
+        p1 = re.compile(r'(?P<number>(\d+)) +(?P<description>(\S+)) +(?P<state>(\S+)) +(?P<model>(\S+)) +(?P<serial>(\S+))$')
+
+        for line in out.splitlines():
+            line = line.strip()
+            m = p1.match(line)
```
Comment (danielgraziano) [inline src/genie/libs/parser/nxos/show_fex.py:50]: same thing about comment showing what the regex matches

**Review by danielgraziano — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/nxos/show_fex.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    def cli(self, output=None):
+        if output is None:
+            # Get output from device
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output 
+        # Initial return dictionary
+        parsed_dict = {}
+
+        p1 = re.compile(r'(?P<number>(\d+)) +(?P<description>(\S+)) +(?P<state>(\S+)) +(?P<model>(\S+)) +(?P<serial>(\S+))$')
```
[the remark this answers is not in the document]
Comment (virrsa) [reply src/genie/libs/parser/nxos/show_fex.py:48]: Added comment in commit 5261d03!

File: src/genie/libs/parser/nxos/show_fex.py (hunk 2 above)
Comment (virrsa) [reply src/genie/libs/parser/nxos/show_fex.py:56]: Same as above, comments were added in 5261d03.

**Review by danielgraziano — APPROVED** [review]

**Review by GerriorL — APPROVED** [review]

**Pull request merged**